### DISCUSSION TITLE
Yet another problem with __tests__ folder inside pages

### DISCUSSION THREAD

**Discussion by angelod1as** [vercel/next.js#25367] [Ideas]
### Describe the feature you'd like to request

It would be really great if Next ignored `__tests__` folder inside `pages` folder. I just lost a few hours of work trying to understand why my project wouldn't build to finally find out it's because I tried _organizing_ my tests inside a folder.

It's not like it's a new request. Like, #1914 #988 #1545 #1689 right?

### Describe the solution you'd like

The `__tests__` folder should be automatically ignored when building.

A better solution would be ignore `*.{specs|test}.{ts,js,tsx,jsx}` too.

### Describe alternatives you've considered

I thought about overriding Next's configuration but this seems a lot of work for something that should come configured out of the box.

Or, at least, give me a better error than `ModuleNotFoundError: Module not found: Error: Can't resolve 'fs' in '/home/angelo/Documentos/GIT/personal/portfolio/node_modules/css/lib/stringify'` when I have a `__tests__` folder inside `pages`.

**Comment by ljosberinn** [comment]
- `pages` is an established special folder in Next, so arguably `__tests__` or anything else that isn't a page isn't supposed to be in there
- all issues you referenced are from 2017
- in my experience, it's also an established pattern to have _one_ `__tests__` folder, not multiple in different places
- you can and probably should skipping transpilation of any test files, as it speeds up the build process (depending on your test amount of course), here's an example for `next.config.js`:
```js
module.exports = {
  webpack: (config, options) {
    config.plugins.push(new options.webpack.IgnorePlugin(/\/__tests__\//));
    return config;
  }
}
```

**Reply by dreamawakening** [reply]
@ljosberinn that Webpack config doesn't work for any tests in `pages`. 

> in my experience, it's also an established pattern to have one `__tests__` folder, not multiple in different places

I strongly disagree. Test files should be next to the file it's testing. Otherwise the global `__tests__` folder ends up having to replicate the file structure for all pages in the `pages` directory.

And regardless, this shouldn't be enforced. It's up to the developer how they want to structure their tests. No other framework I've used enforces putting tests in the root directory.
